examples/notebooks/plot_grid_search_census.py

```diff
@@ -0,0 +1,216 @@
+
+sweep = GridSearch(LogisticRegression(solver='liblinear', fit_intercept=True),
+                   constraints=DemographicParity(),
+                   grid_size=71)
+
+# %%
+# Our algorithms provide :code:`fit()` and :code:`predict()` methods, so they behave in a similar manner
+# to other ML packages in Python.
+# We do however have to specify two extra arguments to :code:`fit()` - the column of protected
+# attribute labels, and also the number of predictors to generate in our sweep.
+#
+# After :code:`fit()` completes, we extract the full set of predictors from the
+# :class:`fairlearn.reductions.GridSearch` object.
+
+sweep.fit(X_train, Y_train,
+          sensitive_features=A_train)
+
+predictors = sweep.predictors_
+
+# %%
+# We could load these predictors into the Fairness dashboard now.
+# However, the plot would be somewhat confusing due to their number.
+# In this case, we are going to remove the predictors which are dominated in the
+# error-disparity space by others from the sweep (note that the disparity will only be
+# calculated for the protected attribute; other potentially protected attributes will
+# not be mitigated).
+# In general, one might not want to do this, since there may be other considerations
+# beyond the strict optimisation of error and disparity (of the given protected attribute).
+
+errors, disparities = [], []
+for m in predictors:
+    def classifier(X): return m.predict(X)
+
+    error = ErrorRate()
+    error.load_data(X_train, pd.Series(Y_train), sensitive_features=A_train)
+    disparity = DemographicParity()
+    disparity.load_data(X_train, pd.Series(Y_train), sensitive_features=A_train)
+
+    errors.append(error.gamma(classifier)[0])
+    disparities.append(disparity.gamma(classifier).max())
+
+all_results = pd.DataFrame({"predictor": predictors, "error": errors, "disparity": disparities})
+
+non_dominated = []
+for row in all_results.itertuples():
+    errors_for_lower_or_eq_disparity = all_results["error"][all_results["disparity"] <= row.disparity]
+    if row.error <= errors_for_lower_or_eq_disparity.min():
+        non_dominated.append(row.predictor)
+
+# %%
+# Finally, we can put the dominant models into the Fairness dashboard, along with the
+# unmitigated model.
+
+dashboard_predicted = {"unmitigated": unmitigated_predictor.predict(X_test)}
+for i in range(len(non_dominated)):
+    key = "dominant_model_{0}".format(i)
+    value = non_dominated[i].predict(X_test)
+    dashboard_predicted[key] = value
+
+
+FairlearnDashboard(sensitive_features=A_test, sensitive_feature_names=['sex'],
+                   y_true=Y_test,
+                   y_pred=dashboard_predicted)
+
+# %%
+# We see a Pareto front forming - the set of predictors which represent optimal tradeoffs
+# between accuracy and disparity in predictions.
+# In the ideal case, we would have a predictor at (1,0) - perfectly accurate and without
+# any unfairness under demographic parity (with respect to the protected attribute "sex").
```

Comment:
here and everywhere else it should be "sensitive feature" instead of "protected attribute"

Reply:
Good point. This obviously hasn't been redrafted since we settled on the naming.